### PR TITLE
Add types to methods and properties batch 2

### PR DESCRIPTION
**Pull request by fbouynot**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
Type hinting random stuff will be very very hard to validate, since if something is wrong, the code just stops with an error there.
The tests doesn't cover anywhere close to everything that needs to be checked.

Therefore each PR should be contained to a specific feature

File: app/Actions/Alerts/RunAlertRulesAction.php [hunk 1]
```diff
@@ -30,14 +30,8 @@
...
-     */
-    private $rules;
-    /**
-     * @var \App\Models\Device
-     */
-    private $device;
+    private AlertRules $rules;
+    private Device $device;
 
     public function __construct(Device $device, AlertRules $rules)
```
Comment (murrant) [inline app/Actions/Alerts/RunAlertRulesAction.php:36]: There is an even better way.  If you define it like this, you can remove the properties and setting them in the constructor completely.
```php
public function __construct(
    private Device $device,
    private AlertRules $rules
)
```

Comment (murrant) [reply app/Actions/Alerts/RunAlertRulesAction.php:36]: My ide informs me they can be readonly too:
```php
    public function __construct(
        private readonly Device $device,
        private readonly AlertRules $rules
    ) {
    }
```

Comment (murrant) [reply app/Actions/Alerts/RunAlertRulesAction.php:36]: And I tend to set readonly properties as public so:
```php
    public function __construct(
        public readonly Device $device,
        public readonly AlertRules $rules
    ) {
    }
```

**Comment by Jellyfrog** [comment]
Another thing, we could mass-typehint automatically, so not sure how worth spending a lot of time doing manually is

If you feel like cleaning up, these are better to start with:
Converting to Eloquent: https://github.com/librenms/librenms/blob/master/phpstan-baseline-deprecated.neon
And the none type hint ones:
```
$ grep -F "message: " phpstan-baseline.neon |  grep -Fv " type specified"                                                                                                                      message: "#^Variable \\$transport in empty\\(\\) always exists and is always falsy\\.$#"
                        message: "#^Static property LibreNMS\\\\Alert\\\\Transport\\\\Sensu\\:\\:\\$severity is never read, only written\\.$#"
                        message: "#^If condition is always false\\.$#"
                        message: "#^Parameter \\#2 \\$password of method LibreNMS\\\\Authentication\\\\MysqlAuthorizer\\:\\:addUser\\(\\) expects string, null given\\.$#"
                        message: "#^Parameter \\#1 \\$string of function str_pad expects string, int given\\.$#"
                        message: "#^Parameter \\#1 \\$key of method Illuminate\\\\Support\\\\Collection\\<int,mixed\\>\\:\\:get\\(\\) expects int, string given\\.$#"
                        message: "#^If condition is always true\\.$#"
                        message: "#^Negated boolean expression is always false\\.$#"
                        message: "#^Property LibreNMS\\\\Data\\\\Store\\\\Rrd\\:\\:\\$async_process \\(LibreNMS\\\\Proc\\) in isset\\(\\) is not nullable\\.$#"
                        message: "#^Property LibreNMS\\\\Data\\\\Store\\\\Rrd\\:\\:\\$sync_process \\(LibreNMS\\\\Proc\\) in isset\\(\\) is not nullable\\.$#"
                        message: "#^Right side of && is always true\\.$#"
                        message: "#^Parameter \\#1 \\$model of static method LibreNMS\\\\Model\\:\\:onCreate\\(\\) expects static\\(LibreNMS\\\\Device\\\\Processor\\), LibreNMS\\\\Device\\\\Processor given\\.$#"
                        message: "#^Parameter \\#1 \\$model of static method LibreNMS\\\\Model\\:\\:onDelete\\(\\) expects static\\(LibreNMS\\\\Device\\\\Processor\\), LibreNMS\\\\Device\\\\Processor given\\.$#"
                        message: "#^Unsafe call to private method LibreNMS\\\\Device\\\\Processor\\:\\:pollProcessors\\(\\) through static\\:\\:\\.$#"
                        message: "#^Unsafe call to private method LibreNMS\\\\Device\\\\Processor\\:\\:processData\\(\\) through static\\:\\:\\.$#"
                        message: "#^Variable \\$adjacencies in empty\\(\\) always exists and is not falsy\\.$#"
                        message: "#^Variable \\$lsps might not be defined\\.$#"
                        message: "#^Variable \\$paths might not be defined\\.$#"
                        message: "#^Variable \\$sdps might not be defined\\.$#"
                        message: "#^Variable \\$svcs might not be defined\\.$#"
                        message: "#^Variable \\$features on left side of \\?\\? is never defined\\.$#"
                        message: "#^Variable \\$hardware on left side of \\?\\? is never defined\\.$#"
                        message: "#^Variable \\$location in empty\\(\\) always exists and is always falsy\\.$#"
                        message: "#^Variable \\$serial on left side of \\?\\? is never defined\\.$#"
                        message: "#^Variable \\$version on left side of \\?\\? is never defined\\.$#"
                        message: "#^If condition is always true\\.$#"
                        message: "#^Property App\\\\Models\\\\Device\\:\\:\\$version \\(string\\|null\\) does not accept int\\.$#"
                        message: "#^Parameter \\#8 \\$multiplier of class LibreNMS\\\\Device\\\\WirelessSensor constructor expects int, null given\\.$#"
                        message: "#^Variable \\$sensors in empty\\(\\) always exists and is not falsy\\.$#"
                        message: "#^Variable \\$multi_get_array on left side of \\?\\? is never defined\\.$#"
                        message: "#^Parameter \\#8 \\$warn_percent of static method LibreNMS\\\\Device\\\\Processor\\:\\:discover\\(\\) expects int, null given\\.$#"
                        message: "#^Parameter \\#8 \\$multiplier of class LibreNMS\\\\Device\\\\WirelessSensor constructor expects int, null given\\.$#"
                        message: "#^Parameter \\#16 \\$entPhysicalIndex of class LibreNMS\\\\Device\\\\WirelessSensor constructor expects float\\|int\\|null, string given\\.$#"
                        message: "#^If condition is always false\\.$#"
                        message: "#^Parameter \\#8 \\$warn_percent of static method LibreNMS\\\\Device\\\\Processor\\:\\:discover\\(\\) expects int, null given\\.$#"
                        message: "#^Variable \\$index might not be defined\\.$#"
                        message: "#^Offset 'sonicCurrentConnCac…' does not exist on string\\.$#"
                        message: "#^Offset 'sonicMaxConnCacheEn…' does not exist on string\\.$#"
                        message: "#^Parameter \\#8 \\$multiplier of class LibreNMS\\\\Device\\\\WirelessSensor constructor expects int, string given\\.$#"
                        message: "#^Parameter \\#9 \\$divisor of class LibreNMS\\\\Device\\\\WirelessSensor constructor expects int, string given\\.$#"
                        message: "#^Variable \\$mplsTunnelARHopIndex in isset\\(\\) always exists and is not nullable\\.$#"
                        message: "#^Variable \\$mplsTunnelARHopListIndex in isset\\(\\) always exists and is not nullable\\.$#"
                        message: "#^Variable \\$sdp_oid in isset\\(\\) always exists and is not nullable\\.$#"
                        message: "#^Variable \\$svc_oid in isset\\(\\) always exists and is not nullable\\.$#"
                        message: "#^Property LibreNMS\\\\Proc\\:\\:\\$_pipes is never written, only read\\.$#"
                        message: "#^If condition is always false\\.$#"
                        message: "#^Unreachable statement \\- code above always terminates\\.$#"
                        message: "#^Variable \\$rrd_options in empty\\(\\) always exists and is always falsy\\.$#"
                        message: "#^Property LibreNMS\\\\Util\\\\Snmpsim\\:\\:\\$proc \\(LibreNMS\\\\Proc\\) in isset\\(\\) is not nullable\\.$#"
                        message: "#^Unreachable statement \\- code above always terminates\\.$#"
                        message: "#^Anonymous function should return non\\-falsy\\-string but returns string\\.$#"
                        message: "#^Property App\\\\Models\\\\Device\\:\\:\\$authlevel \\('authNoPriv'\\|'authPriv'\\|'noAuthNoPriv'\\) does not accept null\\.$#"
                        message: "#^Unable to resolve the template type TReturn in call to method Illuminate\\\\Support\\\\Collection\\<\\(int\\|string\\),Symfony\\\\Component\\\\Console\\\\Input\\\\InputOption\\>\\:\\:flatMap\\(\\)$#"
                        message: "#^Property App\\\\Console\\\\Commands\\\\InternalHttpRequest\\:\\:\\$app is never read, only written\\.$#"
                        message: "#^Ternary operator condition is always true\\.$#"
                        message: "#^PHPDoc type array of property App\\\\Exceptions\\\\Handler\\:\\:\\$dontReport is not covariant with PHPDoc type array\\<string\\> of overridden property Illuminate\\\\Foundation\\\\Exceptions\\\\Handler\\:\\:\\$dontReport\\.$#"
                        message: "#^Called 'Model\\:\\:make\\(\\)' which performs unnecessary work, use 'new Model\\(\\)'\\.$#"
                        message: "#^Call to function is_null\\(\\) with PDO will always evaluate to false\\.$#"
                        message: "#^Ternary operator condition is always false\\.$#"
                        message: "#^Variable \\$vars on left side of \\?\\? is never defined\\.$#"
                        message: "#^Called 'Model\\:\\:make\\(\\)' which performs unnecessary work, use 'new Model\\(\\)'\\.$#"
                        message: "#^Call to an undefined static method App\\\\Models\\\\ServiceTemplate\\:\\:hasAccess\\(\\)\\.$#"
                        message: "#^Called 'Model\\:\\:make\\(\\)' which performs unnecessary work, use 'new Model\\(\\)'\\.$#"
                        message: "#^Cannot call method only\\(\\) on array\\<string, string\\>\\.$#"
                        message: "#^Called 'Model\\:\\:make\\(\\)' which performs unnecessary work, use 'new Model\\(\\)'\\.$#"
                        message: "#^Offset 'image_title' on null in empty\\(\\) does not exist\\.$#"
                        message: "#^PHPDoc type array of property App\\\\Http\\\\Kernel\\:\\:\\$middlewarePriority is not covariant with PHPDoc type array\\<string\\> of overridden property Illuminate\\\\Foundation\\\\Http\\\\Kernel\\:\\:\\$middlewarePriority\\.$#"
                        message: "#^Cannot call method getName\\(\\) on object\\|string\\.$#"
                        message: "#^Cannot call method parameter\\(\\) on object\\|string\\.$#"
                        message: "#^Cannot access property \\$password on object\\|string\\.$#"
                        message: "#^Property App\\\\Jobs\\\\PingCheck\\:\\:\\$deferred is never written, only read\\.$#"
                        message: "#^Property App\\\\Jobs\\\\PingCheck\\:\\:\\$devices \\(Illuminate\\\\Database\\\\Eloquent\\\\Collection\\) in isset\\(\\) is not nullable\\.$#"
                        message: "#^Property App\\\\Jobs\\\\PingCheck\\:\\:\\$devices is never written, only read\\.$#"
                        message: "#^Property App\\\\Jobs\\\\PingCheck\\:\\:\\$groups is never read, only written\\.$#"
                        message: "#^Property App\\\\Jobs\\\\PingCheck\\:\\:\\$tiered is never written, only read\\.$#"
                        message: "#^Unreachable statement \\- code above always terminates\\.$#"
                        message: "#^Ternary operator condition is always true\\.$#"
                        message: "#^Access to an undefined property App\\\\Models\\\\DeviceRelatedModel\\:\\:\\$device_id\\.$#"
                        message: "#^PHPDoc type array of property App\\\\Models\\\\UserPref\\:\\:\\$primaryKey is not covariant with PHPDoc type string of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Model\\:\\:\\$primaryKey\\.$#"
                        message: "#^Result of && is always false\\.$#"
                        message: "#^Right side of && is always false\\.$#"
                        message: "#^Negated boolean expression is always false\\.$#"
                        message: "#^Variable \\$username on left side of \\?\\? is never defined\\.$#"
                        message: "#^Property App\\\\View\\\\SimpleTemplate\\:\\:\\$callback \\(callable\\) on left side of \\?\\? is not nullable\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\AlertScheduleFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\AlertSchedule\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\AlertSchedule\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\BgpPeerFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\BgpPeer\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\BgpPeer\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\BillFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Bill\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Bill\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\ComponentFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Component\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Component\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\DeviceFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Device\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Device\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\DeviceGroupFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\DeviceGroup\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\DeviceGroup\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\Ipv4AddressFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Ipv4Address\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Ipv4Address\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\Ipv4NetworkFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Ipv4Network\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Ipv4Network\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\LocationFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Location\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Location\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\OspfNbrFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\OspfNbr\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\OspfNbr\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\OspfPortFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\OspfPort\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\OspfPort\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\PortFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Port\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Port\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\SensorFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Sensor\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Sensor\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\SyslogFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Syslog\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Syslog\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\UserFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\User\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\User\\>\\:\\:\\$model\\.$#"
                        message: "#^PHPDoc type string of property Database\\\\Factories\\\\VminfoFactory\\:\\:\\$model is not covariant with PHPDoc type class\\-string\\<App\\\\Models\\\\Vminfo\\> of overridden property Illuminate\\\\Database\\\\Eloquent\\\\Factories\\\\Factory\\<App\\\\Models\\\\Vminfo\\>\\:\\:\\$model\\.$#"
                        message: "#^Parameter \\#4 \\$transport of function addHost expects string, int given\\.$#"
                        message: "#^Parameter \\#5 \\$poller_group of function addHost expects string, int given\\.$#"
                        message: "#^Parameter \\#1 \\$attr of method LibreNMS\\\\Authentication\\\\SSOAuthorizer\\:\\:authSSOGetAttr\\(\\) expects string, int given\\.$#"
                        message: "#^Parameter \\#1 \\$attr of method LibreNMS\\\\Authentication\\\\SSOAuthorizer\\:\\:authSSOGetAttr\\(\\) expects string, null given\\.$#"
                        message: "#^Parameter \\#2 \\$default of function set_null expects null, int given\\.$#"
                        message: "#^Parameter \\#1 \\$os of static method LibreNMS\\\\Config\\:\\:getOsSetting\\(\\) expects string, null given\\.$#"
                        message: "#^Negated boolean expression is always false\\.$#"
                        message: "#^Parameter \\#1 \\$new_location of method App\\\\Models\\\\Device\\:\\:setLocation\\(\\) expects App\\\\Models\\\\Location\\|string, null given\\.$#"
```

**Comment by fbouynot** [comment]
> Type hinting random stuff will be very very hard to validate, since if something is wrong, the code just stops with an error there. The tests doesn't cover anywhere close to everything that needs to be checked.
> 
> Therefore each PR should be contained to a specific feature

Alright, should I remove this PR, and start again piece by piece, with a specific piece of code / feature each time, or should I do this for the next one?
I'm fine if with both.

**Comment by fbouynot** [comment]
> Another thing, we could mass-typehint automatically, so not sure how worth spending a lot of time doing manually is
> 
> If you feel like cleaning up, these are better to start with: Converting to Eloquent: 

Oh alright then, I'll let you mass-typehint, and look for other things to do. I'll check if I understand Eloquent parts.

**Comment by murrant** [comment]
Thanks @fbouynot appreciate your PRs.  Closing this one for now.

**Pull request closed without merging**